File: ios/Sources/RynBridgeAuth/AuthModule.swift
import Foundation
import RynBridge

public struct AuthModule: BridgeModule, Sendable {
    public let name = "auth"
    public let version = "0.1.0"
    public let actions: [String: ActionHandler]

    public init(provider: AuthProvider) {
        actions = [
            "login": { payload in
                guard let providerName = payload["provider"]?.stringValue else {
                    throw RynBridgeError(code: .invalidMessage, message: "Missing required field: provider")
                }
                let scopes: [String]
                if let arr = payload["scopes"]?.arrayValue {
                    scopes = arr.compactMap { $0.stringValue }
                } else {
                    scopes = []
                }
                let result = try await provider.login(provider: providerName, scopes: scopes)
                return result.toPayload()
            },
            "logout": { _ in
                try await provider.logout()
                return [:]
            },
            "getToken": { _ in
                let result = try await provider.getToken()
                return result.toPayload()
            },
            "refreshToken": { _ in
                let result = try await provider.refreshToken()
                return result.toPayload()
            },
            "getUser": { _ in
                let user = try await provider.getUser()
                return ["user": user.map { .dictionary($0.toPayload()) } ?? .null]
            },
        ]
    }
}

File: android/auth/src/main/kotlin/com/devryner/rynbridge/auth/AuthModule.kt
package com.devryner.rynbridge.auth

import com.devryner.rynbridge.core.*

class AuthModule(provider: AuthProvider) : BridgeModule {

    override val name = "auth"
    override val version = "0.1.0"
    override val actions: Map<String, ActionHandler> = mapOf(
        "login" to { payload ->
            val providerName = payload["provider"]?.stringValue
                ?: throw RynBridgeError(code = ErrorCode.INVALID_MESSAGE, message = "Missing required field: provider")
            val scopes = payload["scopes"]?.arrayValue
                ?.mapNotNull { it.stringValue }
                ?: emptyList()
            val result = provider.login(providerName, scopes)
            result.toPayload()
        },
        "logout" to { _ ->
            provider.logout()
            emptyMap()
        },
        "getToken" to { _ ->
            val result = provider.getToken()
            result.toPayload()
        },
        "refreshToken" to { _ ->
            val result = provider.refreshToken()
            result.toPayload()
        },
        "getUser" to { _ ->
            val user = provider.getUser()
            mapOf("user" to (user?.let { BridgeValue.dict(it.toPayload()) } ?: BridgeValue.nullValue()))
        }
    )
}
